Assets/MRTK/Providers/Oculus/XRSDK/OculusXRSDKDeviceManager.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.XRSDK.Input;
+using System;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.Oculus
+{
+    /// <summary>
+    /// Manages XR SDK devices on the Oculus platform.
+    /// </summary>
+    [MixedRealityDataProvider(
+        typeof(IMixedRealityInputSystem),
+        SupportedPlatforms.WindowsStandalone | SupportedPlatforms.Android,
+        "XRSDK Oculus Device Manager")]
+    public class OculusXRSDKDeviceManager : XRSDKDeviceManager
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="inputSystem">The <see cref="Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputSystem"/> instance that receives data from this provider.</param>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public OculusXRSDKDeviceManager(
+            IMixedRealityInputSystem inputSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile) { }
+
+        #region Controller Utilities
+
+        /// <inheritdoc />
+        protected override Type GetControllerType(SupportedControllerType supportedControllerType)
+        {
+            switch (supportedControllerType)
+            {
+                case SupportedControllerType.OculusTouch:
+                    return typeof(OculusXRSDKTouchController);
+                default:
+                    return base.GetControllerType(supportedControllerType);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override InputSourceType GetInputSourceType(SupportedControllerType supportedControllerType)
+        {
+            switch (supportedControllerType)
+            {
+                case SupportedControllerType.OculusTouch:
+                    return InputSourceType.Controller;
+                default:
+                    return base.GetInputSourceType(supportedControllerType);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override SupportedControllerType GetCurrentControllerType(InputDevice inputDevice)
+        {
+            if (inputDevice.characteristics.HasFlag(InputDeviceCharacteristics.HandTracking))
+            {
+                if (inputDevice.characteristics.HasFlag(InputDeviceCharacteristics.Left) ||
+                    inputDevice.characteristics.HasFlag(InputDeviceCharacteristics.Right))
+                {
+                    // If it's a hand with a reported handedness, assume articulated hand
+                    return SupportedControllerType.ArticulatedHand;
```

Comment:
It doesn't look like these cases are currently used (and I don't think GGVHand will be needed even when the Oculus XR SDK plug-in supports hand tracking). Should they be removed until it's implemented?